app/handler/provider_complaint.py

```diff
@@ -58,6 +118,32 @@ def get_original_message(cls, message: Message) -> Optional[Message]:
                 return part
         return None
 
+    @classmethod
+    def get_feedback_report(cls, message: Message) -> Optional[Message]:
+        """
+        Find a report that yahoo embeds in the complaint. It has content type 'message/feedback-report'
+        """
+        for part in message.walk():
+            if part["content-type"] == "message/feedback-report":
+                content = part.get_payload()
+                if not content:
+                    continue
+                return content[0]
+        return None
+
+    @classmethod
+    def get_original_addresses(
+        cls, message: Message
+    ) -> Optional[OriginalMessageInformation]:
+        """
+        Try to get the proper recipient from the report that yahoo adds as a port of the complaint. If we cannot find
+        the rcpt in the report or we can't find the report, use the first address in the original message from
+        """
+        report = cls.get_feedback_report(message)
+        original = cls.get_original_message(message)
+        rcpt_header = report["original-rcpt-to"]
```

Comment:
should we create a constant for "original-rcpt-to" in headers.py?

Reply:
I was a bit on the fence with this one since it's a yahoo complaint specific header.. but you're right. better to reduce magic strings. I've set it as a constant in headers.py.